ISICbalanceExtension/TodayViewController.swift

```diff
@@ -0,0 +1,134 @@
+
+    // MARK: - Initialization
+
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        viewModel = TodayViewModel(dependencies: AppDependency.shared)
+        
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Controller lifecycle
+
+    override func loadView() {
+        super.loadView()
+
+        let balanceStack = UIStackView()
+        balanceStack.axis = .vertical
+        balanceStack.alignment = .center
+        balanceStack.spacing = 5
+        balanceStack.sizeToFit()
+        view.addSubview(balanceStack)
+
+        let balanceTitle = UILabel()
+        // we are not able to use swiftgen, because of retain cycle
+        balanceTitle.text = NSLocalizedString("balance.title", comment: "ISIC account balance")
+        self.balanceTitle = balanceTitle
+        balanceStack.addArrangedSubview(balanceTitle)
+
+        let balanceLabel = UILabel()
+        balanceLabel.font = UIFont.boldSystemFont(ofSize: 30)
+        self.balanceLabel = balanceLabel
+        balanceStack.addArrangedSubview(balanceLabel)
+
+        balanceStack.snp.makeConstraints { make in
+            make.leading.equalToSuperview().inset(50)
+            make.top.greaterThanOrEqualToSuperview()
+            make.bottom.lessThanOrEqualToSuperview()
+            make.centerY.equalToSuperview()
+        }
+
+        let isicImage = UIImage(named: "todayExtension")
+        let appButton = UIButton(type: .custom)
+        appButton.setImage(isicImage, for: .normal)
+        self.appButton = appButton
+        view.addSubview(appButton)
+
+        appButton.snp.makeConstraints { make in
+            make.leading.greaterThanOrEqualTo(balanceStack.snp.trailing).offset(20)
+            make.trailing.equalToSuperview().inset(50)
+            make.top.bottom.equalToSuperview()
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+        appButton?.addTarget(self, action: #selector(isicImageTapped), for: .touchUpInside)
+        setupBindings()
+    }
+
+    // "This method is called to give a widget an opportunity to update its content and redraw its view prior to an operation such as a snapshot"
+    func widgetPerformUpdate(completionHandler: (@escaping (NCUpdateResult) -> Void)) {
+        // Perform any setup necessary in order to update the view.
+
+        // If an error is encountered, use NCUpdateResult.Failed
+        // If there's no update required, use NCUpdateResult.NoData
+        // If there's an update, use NCUpdateResult.NewData
+        viewModel.actions.getBalance.apply().start()
+
+        completionHandler(NCUpdateResult.newData)
+    }
+
+    // MARK: - Bindings
+
+    private func setupBindings() {
+        balanceLabel?.reactive.text <~ viewModel.localeBalance
```

Comment:
❔ proč tady může být otazník, když je to nahoře zadefinováno jako `UILabel`? A jaký to má rozdíl proti tomu, když tam ten otazník nebude?